insights.go

```diff
@@ -21,29 +18,18 @@ func unregisterInsights() error {
 }
 
 func insightsIsRegistered() (bool, error) {
-	var errBuffer bytes.Buffer
-	cmd := exec.Command("/usr/bin/insights-client", "--status")
-	cmd.Stderr = &errBuffer
-
-	err := cmd.Run()
-
+	// While `insights-client --status` properly checks for registration status by
+	// asking Inventory, its two modes (legacy v. non-legacy API) behave
+	// differently (they return different texts with different exit codes).
+	// The `.registered` file is always present on the registered system.
+	_ = exec.Command("/usr/bin/insights-client", "--status").Run()
```

Comment:
I would not ignore returned error here. E.g. If the file `/usr/bin/insights-client` did not exist, then I would return such error.